packages/dcos-sdk/src/RoutingService.ts

```diff
@@ -0,0 +1,43 @@
+import { injectable, inject, named } from "inversify";
+import { EventEmitter } from "events";
+import { IExtensionProvider, ExtensionProvider } from "plugin-baby";
+import { IPartialNextState } from "react-router"
+
+const ROUTING_CHANGED = Symbol("ROUTING_CHANGED");
+
+export const RoutingServiceExtension = Symbol("RoutingServiceExtension");
+
+export interface IRoutingExtension {
+  getRoutes(partialNextState: IPartialNextState): any[];
+}
+
+@injectable()
+export default class RoutingService {
+  private extensionProvider: IExtensionProvider<IRoutingExtension>;
+  private eventEmmiter: EventEmitter;
+
+  constructor(
+    @inject(ExtensionProvider)
+    @named(RoutingServiceExtension)
+    extensionProvider: IExtensionProvider<IRoutingExtension>
+  ) {
+    this.extensionProvider = extensionProvider;
+    this.eventEmmiter = new EventEmitter();
+  }
+
+  public on(type: symbol, callback: () => void) {
+    this.eventEmmiter.on(type, callback);
+  }
+
+  public getRoutes(partialNextState: IPartialNextState) {
+    return this.extensionProvider
+      .getAllExtensions()
+      .reduce((acc: any[], extension: IRoutingExtension) =>
```

Comment:
shouldn't the type of acc be a list of IRoutingExtension?

Reply:
It returns a list of react-router routes

Reply:
I think I meant something like:

```ts
.reduce((acc: IRoutingExtension[], extension: IRoutingExtension) =>
```

Reply:
Like I said, reduce accumulator accumulates Routes
```ts
.reduce((acc: Route[], extension: IRoutingExtension) =>
```

Reply:
Yes, that is what I meant :D. Thanks for the correction.